ui/uistatemanager.cpp

```diff
@@ -167,10 +173,32 @@ void UIStateManager::restoreState()
     restoreWindowState();
     restoreSplitterState();
     restoreHeaderState();
+
+    // Allow custom state per end point
+#if QT_VERSION >= QT_VERSION_CHECK(5, 0, 0)
+    if (m_customRestoreMethod.isValid())
+#endif
```

Comment:
Does this mean we end up calling invalid methods with Qt4? Might be safer to entirely skip this with Qt4 in that case? (same again below)